### ISSUE TITLE
[improvement] UX improvement proposal

### ISSUE THREAD

**Issue by wujekbogdan** [polylang/polylang#161]
A long time ago I've posted [a topic on wordpress.org](https://wordpress.org/support/topic/ideas-how-to-improve-the-uiux/). But there are so many things happening there, that it probably got forgotten. 

So let me post it here, on Github again.

---

First of all thanks for the plugin. It’s much better and cleaners solution than WPML and it works great with ACF out of the box. For me it’s perfect, but I found it difficult to use for end-users.

I’ll try to show you some usability issues and try to show some solutions as well.

First of all average users are not familiar with WordPress data structure, they don’t get the idea of posts, and they don’t understand that in fact, each translation is just another post. And because of this, they don't understand how translations plugins (like Polylang or WPML work). They just want to translate posts.

**Let’s imagine how an average user tries to use this plugin:**

- [This is what a user sees](https://user-images.githubusercontent.com/533954/29316360-a21c7876-81bf-11e7-97bd-a5b6148a7bac.png) when the page is not translated yet. What’s a first thing what he tries to do? There is a text input and a pencil icon. User tries to type something in the text input. He’s not aware yet it’s not a regular input, but a search field. So he tries to type a new post name there. **Yes, this is what the majority of users do!**
- He doesn’t notice the loader that shows up for a while (or he just ignores it). Nothing special has happened when he typed the text so he clicks the edit button.
- Now the user is redirected to a new page with a blank title field. It’s strange for him because he has just entered a new page name in the text field (on the previous page). He ignores this fact. It’s a new tool. Strange things are happening. He might think it’s a bug, or maybe he did something wrong. Whatever.
- So the new page has been created. The user is a bit confused. He’s not sure what has just happened. Did he just create a translation, or maybe something went wrong?

This scenario is based on my own experience and by my client's experience. Each time when it comes to WP posts translation I have to guide my clients through the process and explain the idea that's behind the Polylang plugin very carefully. But even smart ones have some problems. The majority of problems are caused by the plugins users interface, which can be improved without a great effort. These are just little details that can improve the workflow a lot.

I have some ideas how to improve the workflow:

---
Here's the mockup:

![polylang](https://user-images.githubusercontent.com/533954/29316249-2d609aee-81bf-11e7-8f22-56ac08cff02e.png)

I’m not 100% sure about a single button next to the language selection input. Maybe it would be better to show 2 buttons: one to assign a translation and another one to create a new page?

Metabox:
<img width="297" alt="polylang-metabox-new" src="https://user-images.githubusercontent.com/533954/29316286-580c018e-81bf-11e7-8e1c-5671813355cb.png">

**Comment by grappler** [comment]
This is really interesting.

Adding a description and more text to the drop down seems like a good idea.

What I am not sure is about the extra buttons. In WordPress there are no extra buttons for updating the single fields as all of the fields are updated with the update button.

The suggested way of creating a translation would not be clear to me that I would have to first add a possible title name in  the field before I could create a translation.

One idea that I had was to hide the search field and just have two buttons "Create" / "Select". The problem that I see with this is when connecting a translation it requires an extra step/click.

Related to this meta box is the idea of adding a link to the translation posts in #147

**Comment by wujekbogdan** [comment]
> What I am not sure is about the extra buttons. In WordPress there are no extra buttons for updating the single fields as all of the fields are updated with the update button.

Yeah, you might be right. I do not insist to add these buttons... on the other hand, I can't agree that there are no extra buttons on the post edit page. There is a post slug edit button, there are buttons for adding custom fields (native WP custom fields, or extra buttons added by plugins like ACF). But these buttons do not update a post once they're pressed. A post is saved once a post save button is clicked. 

---
I'm sure that the mockup that I've prepared is not perfect. It's more a concept than a solution that's ready for implementation.

The main goal of this task is to explain the workflow issues.

**Comment by Chouby** [comment]
@wujekbogdan I remember the topic. Although I worked on totally different issues, I had kept a copy of your mockup on my hard disk. It's nice that you opened the issue here as it will make it more visible to get valuable feedback from other users.

**Comment by wujekbogdan** [comment]
@Chouby 
>  I remember the topic. Although I worked on totally different issues, I had kept a copy of your mockup on my hard disk.

Great. I thought it was forgotten :)

> It's nice that you opened the issue here as it will make it more visible to get valuable feedback from other users

I can try to implement it and create a pull request. But I can't promise. It depends on how much free time I will have.

**Comment by kLOsk** [comment]
Without providing a solution I want to add that ~~almost~~ ALL of my clients make the mistake and changing the language of a post using the language dropdown selector. With one client I even went that far and applied css to hide the dropdown! I believe this is actually the number one UX issue with Polylang at the moment.
